### PR TITLE
[GLUTEN-4479][VL] Map `unbase64` to corr. Presto function

### PR DESCRIPTION
**Pull request by sujithjay**
## Changes Proposed in this Pull Request

This PR maps `unbase` Spark function to  the`from_base64` function from Presto.

(Fixes: #4479)

## How Was This Patch Tested?

This was tested using a simple query `select unbase64(e) from t`.

NOTE: This PR is based on @fyp711's work in https://github.com/apache/incubator-gluten/pull/4482, with minor changes.

### CODE REVIEW

**Comment by PHILO-HE** [comment]
Hi @sujithjay, we are removing presto scalar function registry in Gluten to fix conflict issues, see https://github.com/apache/incubator-gluten/pull/5202. So could you register this presto function for sparksql in velox code? Generally, it just requires to include the corresponding header, then register it with a name aligned with Spark.

**Comment by fyp711** [comment]
Looks good!

**Comment by sujithjay** [comment]
> Hi @sujithjay, we are removing presto scalar function registry in Gluten to fix conflict issues, see #5202. So could you register this presto function for sparksql in velox code? Generally, it just requires to include the corresponding header, then register it with a name aligned with Spark.

Hi @PHILO-HE, please review the latest commit (https://github.com/apache/incubator-gluten/pull/5242/commits/40747a67638b7d7257b1549538c3da43cf0c8889). Is this what you meant? Let me know if I am missing something. Thank you.

**Comment by PHILO-HE** [comment]
> > Hi @sujithjay, we are removing presto scalar function registry in Gluten to fix conflict issues, see #5202. So could you register this presto function for sparksql in velox code? Generally, it just requires to include the corresponding header, then register it with a name aligned with Spark.
> 
> Hi @PHILO-HE, please review the latest commit ([40747a6](https://github.com/apache/incubator-gluten/commit/40747a67638b7d7257b1549538c3da43cf0c8889)). Is this what you meant? Let me know if I am missing something. Thank you.

@sujithjay, let me clarify a bit. I meant we need to change some Velox code, i.e., add this presto function into sparksql's registry (register it with "unbase64" name). Here is an example: https://github.com/facebookincubator/velox/pull/9215/files

After that, Gluten will not need mapping  "unbase64" to presto function name. Thanks!

File: gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala [hunk 1]
```diff
@@ -144,6 +144,7 @@ class VeloxTestSettings extends BackendTestSettings {
   enableSuite[GlutenRegexpExpressionsSuite]
   enableSuite[GlutenSortOrderExpressionsSuite]
   enableSuite[GlutenStringExpressionsSuite]
+    .exclude("base64/unbase64 for string")
```
Comment (PHILO-HE) [inline gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala:147]: Could you clarify why we need to remove this test? Thanks!

Comment (fyp711) [reply gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala:147]: > Could you clarify why we need to remove this test? Thanks!

@sujithjay @PHILO-HE There seems some differences between the unbase64 function provided by Velox and Spark, such as when the length of the passed string is not a multiple of 4.

Comment (PHILO-HE) [reply gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala:147]: @sujithjay, it would be better to firstly investigate the differences. If only corner cases are involved, we can ignore the difference. Otherwise, better to implement a new Velox function for Spark.

Comment (sujithjay) [reply gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala:147]: I will remove the unnecessary test exclusion.

Comment (fyp711) [reply gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala:147]: @sujithjay It's perfect, Could you do a test? It seems if it's not a multiple of 4 then the spark will return null, but the velox will crash (throw an exception) . The reason for adding this filter is because of this difference. Thanks.

Comment (supermem613) [reply gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala:147]: @fyp711 , @sujithjay , @PHILO-HE, note that there are a few failure mode behavior differences between velox and spark, all boiling down to the java implementation of base64 decoding, so it is a bit more than just returning null vs. exception on non-multiples of 4. Some examples in Spark, gleaned from looking at Java sources:

* Extraneous padding is ignored. For example, "YQ=====" decodes to "a"
* Missing padding is ignored in some cases. For example, "aA" is decoded to "h"
* Invalid characters in some cases are ignored. For example, "@@dmVsb3g=" decodes to "velox".

Comment (PHILO-HE) [reply gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala:147]: @supermem613, thanks a lot for your information! @sujithjay, could you help confirm? We may need to align the implementation with Spark.

Comment (sujithjay) [reply gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala:147]: Thank you @supermem613. If you could share any tests you used to compare these cases, that would be helpful.

Comment (supermem613) [reply gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala:147]: Sure. Here are some:

"YQ=" -> "a"
"YQ===" -> "a"
"" -> ""

"aA" " -> "h"
"c3d" -> "sw"
"cd@" -> "q"

"SGVsbG8gV29ybGQgZnJvbSBWZW@@xveCE=" -> "Hello World from Velox!"
"@@dmVsb3g=" => "velox"

"a" -> exception
"c@" -> exception
"==" -> exception
"SGVsbG8gV29ybGQgZnJvbSBWZW===xveCE=" -> exception

Comment (sujithjay) [reply gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala:147]: @PHILO-HE I confirm that some of the above cases shared by @supermem613 show behavior differences for Spark and Velox. Do you agree that we should implement `base64` and `unbase64` in Velox that behaves the same as Spark? If yes, I will close this PR and start implementing them on Velox.

Comment (PHILO-HE) [reply gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala:147]: @sujithjay, yes, it would be nice to add a new implementation in Velox. Thanks!
